### PR TITLE
fix(compiler): invalid code location in errors when using multibyte chars

### PR DESCRIPTION
**Pull request by yoav-steinberg**
Multi byte chars in the code (mostly in comments) lead to wrong spans being produced for diagnostics:
This test:
```wing
// Here we have some multibyte haiku which we'll use to verify
// the reported diagnostics point to the correct location (line 4 column 1)
// 古池や蛙飛び込む水の音
asdf;
//^^ Unknown symbol "asdf"
```
Produced this wrong output before the fix:
image

We now calculate the byte offset of the diags which is what our diag reporting tool ([codespan-reporting](https://crates.io/crates/codespan-reporting)) expects instead of the character offset.

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributors/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Review by staycoolcall911 — APPROVED** [review]
So that was the bug... nice!

File: examples/tests/invalid/diags_with_multibyte_chars.w [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Here we have some multibyte haiku which we'll use to verify
+// the reported diagnostics point to the correct location (line 4 column 1)
+// 古池や蛙飛び込む水の音
+asdf;
```
[the remark this answers is not in the document]
Comment (MarkMcCulloh) [reply examples/tests/invalid/diags_with_multibyte_chars.w:4]: the lsp tests currently don't deal with multi byte chars, dealing with them there would be a little more annoying

[the remark this answers is not in the document]
Comment (yoav-steinberg) [reply examples/tests/invalid/diags_with_multibyte_chars.w:4]: Right, it's a human snapshot verification test. I was explicit in the comment about what we should expect, hopefully if this changes we'll notice...

**Review by Chriscbr — COMMENTED** [review]
💪

**Pull request merged**